templates/unattended-upgrades.j2

```diff
@@ -30,21 +30,35 @@ APT::Periodic::RandomSleep "{{unattended_random_sleep}}";
 
 
 // Overrides for `50unattended-upgrades`. ////////////////////////////////////
-
-// Unattended-Upgrade::Origins-Pattern controls which packages are
-// upgraded.
-Unattended-Upgrade::Origins-Pattern {
-{% if unattended_origins_patterns is defined %}
-  {% for origin in unattended_origins_patterns %}
-    "{{ origin }}";
-  {% endfor %}
-{% else %}
-  {% for origin in __unattended_origins_patterns %}
-   "{{ origin }}";
-  {% endfor %}
+{% if unattended_origins_patterns is defined or __unattended_origins_patterns is defined %}
+  // Unattended-Upgrade::Origins-Pattern controls which packages are
+  // upgraded.
+  Unattended-Upgrade::Origins-Pattern {
+  {% if unattended_origins_patterns is defined %}
+    {% for origin in unattended_origins_patterns %}
+      "{{ origin }}";
+    {% endfor %}
+  {% else %}
+    {% for origin in __unattended_origins_patterns %}
+     "{{ origin }}";
+    {% endfor %}
+  {% endif %}
+  };
+{% endif %}
+{% if unattended_allowed_origins is defined or __unattended_allowed_origins is defined %}
+  // Automatically upgrade packages from these (origin:archive) pairs
+  Unattended-Upgrade::Allowed-Origins {
+  {% if unattended_allowed_origins is defined %}
+    {% for origin in unattended_origins_patterns %}
+      "{{ origin }}";
+    {% endfor %}
+  {% else %}
+    {% for origin in __unattended_allowed_origins %}
+      "{{ origin }}";
+    {% endfor %}
+  {% endif %}
```

Comment:
This doesn't look quite right. You're checking for one variable, but then looping through another.